### PR TITLE
202109 Runtime logic

### PR DESCRIPTION
**Pull request by owenThurm**
Runtime Logic

### CODE REVIEW

File: src/GUI/bot_gui.py [hunk 1]
```diff
@@ -204,16 +205,36 @@ def comment_round(self) -> None:
             self.targets_index = (self.targets_index + 1) % len(self.targets_list)
             self.save_entries()
 
-            if is_comm_blocked or is_like_blocked:
-                next_run_time = datetime.now() + timedelta(hours=24)
-                self.display_blocked_popup()
+            self.schedule_next_run_time(force_sleep=is_comm_blocked or is_like_blocked)
```
Comment (DanielGelfand) [inline src/GUI/bot_gui.py:208]: Noice.

File: src/GUI/bot_gui.py [hunk 2]
```diff
@@ -204,16 +205,36 @@ def comment_round(self) -> None:
...
+        """
+        Schedules the next time to run by updating self vars and
+        runtime.txt.
+
+        Allows for 8 runs in a day
+        """
+
+        if force_sleep or (self.runs_today >= 8 and randint(0, 13) == 7):
+            # Randomly sleep for a whole day + time until morning
+            # -> rest for 34.5 hours
```
Comment (DanielGelfand) [inline src/GUI/bot_gui.py:221]: 34.5?

File: src/utils.py [hunk 3]
```diff
@@ -22,25 +22,29 @@ def resource_path(relative_path: str) -> str:
...
     """
     Gets the next run time for the bot by reading from runtime.txt.
     If runtime.txt is unavailable or empty, None is returned.
     """
 
     try:
-        return datetime.strptime(
-            open(resource_path(RUNTIME_FILE_PATH), "r").read(), "%Y-%m-%d %H:%M:%S.%f"
-        )
+        run_info = open(resource_path(RUNTIME_FILE_PATH), "r").read()
```
Comment (DanielGelfand) [inline src/utils.py:38]: Looks good

File: src/GUI/bot_gui.py (hunk 2 above)
Comment (owenThurm) [reply src/GUI/bot_gui.py:221]: Forgot to update comment

Comment (owenThurm) [reply src/GUI/bot_gui.py]: Fixed, nice catch

Comment (owenThurm) [reply src/GUI/bot_gui.py]: On another note, I noticed our GUI code is really tied to our qeue/bot execution logic. I think in the future I'll make a PR which separates these into a more MVC approach.

File: src/GUI/bot_gui.py [hunk 4]
```diff
@@ -119,14 +119,14 @@ def get_inputs(self) -> Tuple:
             self.main_entry.get()
             if not is_main
             and self.main_entry.get()
-            and not self.main_entry.get() == SEND_TO_PLACEHOLDER
+            and self.main_entry.get() != SEND_TO_PLACEHOLDER
             else None
         )
-        num_comments = int(self.comment_number_entry.get()) / 10
+        num_comments = int(self.comment_number_entry.get()) // 8
```
Comment (owenThurm) [inline src/GUI/bot_gui.py:125]: 8 runs per day so that they can leave their laptops on for a shorter period

**Review by DanielGelfand — APPROVED** [review]
Logic works

**Pull request merged**